<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>PipeNetworkGUI</class>
 <widget class="QMainWindow" name="PipeNetworkGUI">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>808</width>
    <height>602</height>
   </rect>
  </property>
  <property name="maximumSize">
   <size>
    <width>16777215</width>
    <height>16777215</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Pipe Network Analysis</string>
  </property>
  <property name="autoFillBackground">
   <bool>false</bool>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QHBoxLayout" name="horizontalLayout_2">
    <item>
     <layout class="QVBoxLayout" name="verticalLayout">
      <property name="sizeConstraint">
       <enum>QLayout::SetFixedSize</enum>
      </property>
      <item>
       <widget class="QGraphicsView" name="graphicsView">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="resultsGroupBox">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Minimum">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>90</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>16777215</width>
          <height>80</height>
         </size>
        </property>
        <property name="title">
         <string>Results</string>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_2">
         <item>
          <widget class="QTreeWidget" name="resultsList">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Preferred" vsizetype="Minimum">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="minimumSize">
            <size>
             <width>0</width>
             <height>60</height>
            </size>
           </property>
           <property name="frameShape">
            <enum>QFrame::NoFrame</enum>
           </property>
           <property name="horizontalScrollBarPolicy">
            <enum>Qt::ScrollBarAsNeeded</enum>
           </property>
           <attribute name="headerVisible">
            <bool>true</bool>
           </attribute>
           <attribute name="headerDefaultSectionSize">
            <number>70</number>
           </attribute>
           <attribute name="headerMinimumSectionSize">
            <number>25</number>
           </attribute>
           <attribute name="headerShowSortIndicator" stdset="0">
            <bool>false</bool>
           </attribute>
           <attribute name="headerStretchLastSection">
            <bool>true</bool>
           </attribute>
           <column>
            <property name="text">
             <string>D [cm]</string>
            </property>
           </column>
           <column>
            <property name="text">
             <string>L [m]</string>
            </property>
           </column>
           <column>
            <property name="text">
             <string>Q [L/s]</string>
            </property>
           </column>
           <column>
            <property name="text">
             <string>Î”P [kPa]</string>
            </property>
           </column>
           <column>
            <property name="text">
             <string>k</string>
            </property>
           </column>
           <column>
            <property name="text">
             <string>f</string>
            </property>
           </column>
           <column>
            <property name="text">
             <string>Re</string>
            </property>
           </column>
           <column>
            <property name="text">
             <string>m [kg/s]</string>
            </property>
           </column>
           <column>
            <property name="text">
             <string>V [m/s]</string>
            </property>
           </column>
           <column>
            <property name="text">
             <string>P Inlet [kPa]</string>
            </property>
           </column>
           <column>
            <property name="text">
             <string>P Outlet [kPa]</string>
            </property>
           </column>
           <item>
            <property name="text">
             <string>0.0</string>
            </property>
            <property name="text">
             <string>0.0</string>
            </property>
            <property name="text">
             <string>0.0</string>
            </property>
            <property name="text">
             <string>0.0</string>
            </property>
            <property name="text">
             <string>0.0</string>
            </property>
            <property name="text">
             <string>0.0</string>
            </property>
            <property name="text">
             <string>0.0</string>
            </property>
            <property name="text">
             <string>0.0</string>
            </property>
            <property name="text">
             <string>0.0</string>
            </property>
            <property name="text">
             <string>0.0</string>
            </property>
            <property name="text">
             <string>0.0</string>
            </property>
           </item>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QVBoxLayout" name="verticalLayout_4">
      <item>
       <widget class="QGroupBox" name="groupBox_2">
        <property name="title">
         <string>Add Item</string>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_3">
         <item>
          <widget class="QComboBox" name="addItemIndex">
           <item>
            <property name="text">
             <string>Pipe</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Valve</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Fitting</string>
            </property>
           </item>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="addItemButton">
           <property name="text">
            <string>Add</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="fluidPropertiesGroup">
        <property name="title">
         <string>Fluid</string>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_5">
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout">
           <item>
            <widget class="QComboBox" name="fluidSelection">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <item>
              <property name="text">
               <string>Custom</string>
              </property>
             </item>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_3">
           <item>
            <widget class="QLabel" name="densityLabel">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="text">
              <string>Density</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QDoubleSpinBox" name="customDensityBox">
             <property name="decimals">
              <number>2</number>
             </property>
             <property name="minimum">
              <double>0.010000000000000</double>
             </property>
             <property name="maximum">
              <double>9999.000000000000000</double>
             </property>
             <property name="value">
              <double>998.000000000000000</double>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="densityUnits">
             <property name="text">
              <string>[kg/m^3]</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_5">
           <item>
            <widget class="QLabel" name="viscosityLabel">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="text">
              <string>Viscosity</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QDoubleSpinBox" name="customViscosityBox">
             <property name="decimals">
              <number>2</number>
             </property>
             <property name="maximum">
              <double>9999999.000000000000000</double>
             </property>
             <property name="value">
              <double>1787.000000000000000</double>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="viscosityUnits">
             <property name="text">
              <string>[Î¼Pa*s]</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="shorcutGroupBox">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Minimum">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="title">
         <string>Shorcuts</string>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_6">
         <item>
          <widget class="QPushButton" name="runShortcut">
           <property name="text">
            <string>Run</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="clearAllShorcut">
           <property name="text">
            <string>Clear All</string>
           </property>
          </widget>
         </item>
        </layout>
        <zorder>clearAllShorcut</zorder>
        <zorder>runShortcut</zorder>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>808</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionNew"/>
    <addaction name="actionOpen"/>
    <addaction name="actionSave"/>
    <addaction name="separator"/>
    <addaction name="actionExit"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="separator"/>
    <addaction name="actionAbout"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuHelp"/>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <property name="movable">
    <bool>false</bool>
   </property>
   <property name="allowedAreas">
    <set>Qt::TopToolBarArea</set>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionNew"/>
   <addaction name="actionOpen"/>
   <addaction name="actionSave"/>
   <addaction name="separator"/>
   <addaction name="actionRun"/>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="actionNew">
   <property name="icon">
    <iconset>
     <normaloff>resources/New.png</normaloff>resources/New.png</iconset>
   </property>
   <property name="text">
    <string>New</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+N</string>
   </property>
  </action>
  <action name="actionOpen">
   <property name="icon">
    <iconset>
     <normaloff>resources/Open.png</normaloff>resources/Open.png</iconset>
   </property>
   <property name="text">
    <string>Open</string>
   </property>
   <property name="toolTip">
    <string>Open a saved file</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="icon">
    <iconset>
     <normaloff>resources/Save.png</normaloff>resources/Save.png</iconset>
   </property>
   <property name="text">
    <string>Save</string>
   </property>
   <property name="toolTip">
    <string>Save current configration</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actionExit">
   <property name="text">
    <string>Exit</string>
   </property>
   <property name="toolTip">
    <string>Exit program</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Q</string>
   </property>
  </action>
  <action name="actionAbout">
   <property name="text">
    <string>About</string>
   </property>
  </action>
  <action name="actionRun">
   <property name="text">
    <string>Run</string>
   </property>
   <property name="toolTip">
    <string>Run simulation</string>
   </property>
   <property name="shortcut">
    <string>F5</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections>
  <connection>
   <sender>actionExit</sender>
   <signal>triggered()</signal>
   <receiver>PipeNetworkGUI</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionAbout</sender>
   <signal>triggered()</signal>
   <receiver>PipeNetworkGUI</receiver>
   <slot>showAbout()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionNew</sender>
   <signal>triggered()</signal>
   <receiver>PipeNetworkGUI</receiver>
   <slot>newAction()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionOpen</sender>
   <signal>triggered()</signal>
   <receiver>PipeNetworkGUI</receiver>
   <slot>openAction()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionSave</sender>
   <signal>triggered()</signal>
   <receiver>PipeNetworkGUI</receiver>
   <slot>saveAction()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>clearAllShorcut</sender>
   <signal>clicked()</signal>
   <receiver>PipeNetworkGUI</receiver>
   <slot>clearAll()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>677</x>
     <y>536</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>addItemButton</sender>
   <signal>clicked()</signal>
   <receiver>PipeNetworkGUI</receiver>
   <slot>addItem()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>715</x>
     <y>179</y>
    </hint>
    <hint type="destinationlabel">
     <x>403</x>
     <y>300</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>fluidSelection</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>PipeNetworkGUI</receiver>
   <slot>updateFluidSelection()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>709</x>
     <y>278</y>
    </hint>
    <hint type="destinationlabel">
     <x>403</x>
     <y>300</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>customDensityBox</sender>
   <signal>valueChanged(double)</signal>
   <receiver>PipeNetworkGUI</receiver>
   <slot>updateCustomFluid()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>705</x>
     <y>324</y>
    </hint>
    <hint type="destinationlabel">
     <x>403</x>
     <y>300</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>customViscosityBox</sender>
   <signal>valueChanged(double)</signal>
   <receiver>PipeNetworkGUI</receiver>
   <slot>updateCustomFluid()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>710</x>
     <y>370</y>
    </hint>
    <hint type="destinationlabel">
     <x>403</x>
     <y>300</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>runShortcut</sender>
   <signal>clicked()</signal>
   <receiver>PipeNetworkGUI</receiver>
   <slot>runSimulation()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>709</x>
     <y>467</y>
    </hint>
    <hint type="destinationlabel">
     <x>403</x>
     <y>300</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionRun</sender>
   <signal>triggered()</signal>
   <receiver>PipeNetworkGUI</receiver>
   <slot>runSimulation()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>403</x>
     <y>300</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>newAction()</slot>
  <slot>openAction()</slot>
  <slot>saveAction()</slot>
  <slot>showAbout()</slot>
  <slot>addItem()</slot>
  <slot>clearAll()</slot>
  <slot>updateFluidSelection()</slot>
  <slot>updateCustomFluid()</slot>
  <slot>runSimulation()</slot>
 </slots>
</ui>
